tests/test_notes.py

```diff
@@ -0,0 +1,312 @@
+   minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+   aliquip ex ea commodo consequat. Duis aute irure dolor in
+   reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
+   pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
+   culpa qui officia deserunt mollit anim id est laborum.
+"""
+RELEASE_NOTES_EMPTY = """## release-tools 0.8.10 - (2019-01-01)
+
+No changes list available.
+"""
+RELEASE_NOTES_FILE_ALREADY_EXISTS_ERROR = (
+    "Error: Release notes for version 0.8.10 already exist. Use '--overwrite' to replace it."
+)
+CHANGELOG_DIR_NOT_FOUND = (
+    r"Error: changelog entries directory .+ does not exist"
+)
+INVALID_NAME_ERROR = (
+    "Error: Invalid value for \"NAME\": cannot be empty"
+)
+INVALID_VERSION_ERROR = (
+    "Error: Invalid value for \"VERSION\": cannot be empty"
+)
+
+
+class TestNotes(unittest.TestCase):
+    """Unit tests for notes script"""
+
+    @staticmethod
+    def setup_unreleased_entries(dirpath):
+        """Set up a set of unreleased entry files"""
+
+        titles = [
+            "first feature",
+            "first bug fix",
+            "second feature",
+            "final feature",
+            "another fix",
+        ]
+        categories = [
+            CategoryChange.ADDED,
+            CategoryChange.FIXED,
+            CategoryChange.ADDED,
+            CategoryChange.ADDED,
+            CategoryChange.FIXED,
+        ]
+        authors = ['jsmith', 'jdoe', 'jsmith', 'jdoe', 'jsmith']
+        prs = [1, 2, 3, 'null', 'null']
+        notes = [True, False, False, False, True]
+        notes_txt = (
+            ">\n"
+            "  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor"
+            " incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,"
+            " quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo"
+            " consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse\n"
+            "  cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat"
+            " non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+        )
+
+        os.makedirs(dirpath)
+
+        # Create some entries
+        for x in range(0, 5):
+            filepath = os.path.join(dirpath, str(x) + '.yml')
+
+            with open(filepath, mode='w') as fd:
+                msg = "---\ntitle: {}\ncategory: {}\n"
+                msg += "author: {}\npull_request: {}\nnotes: {}\n"
+                ntxt = notes_txt if notes[x] else "null"
+                msg = msg.format(titles[x], categories[x].category, authors[x],
+                                 prs[x], ntxt)
+                fd.write(msg)
+
+    @unittest.mock.patch('release_tools.notes.ReleaseNotesComposer._datetime_utcnow_str')
+    @unittest.mock.patch('release_tools.notes.determine_changelog_entries_dirpath')
+    def test_release_notes(self, mock_dirpath, mock_utcnow):
```

Comment:
there is no test that executes notes with `dry-run`

Reply:
Fixed.